File: ProtoSASOnScala/src/main/scala/dd/sas/pipeline/calculation/structures/nodes/generativeNodes/InitialGenerativeNode.scala
package dd.sas.pipeline.calculation.structures.nodes.generativeNodes

import dd.sas.pipeline.calculation.structures.Level
import dd.sas.pipeline.calculation.structures.flows.Flow
import dd.sas.pipeline.calculation.structures.flows.generativeflows.{InitialGenerativeFlow, ObjectFlow}
import dd.sas.pipeline.calculation.structures.nodes.Node
import dd.sas.pipeline.worldmodel.WorldObject

/**
  * Created by Sergey on 02.10.2016.
  */
class InitialGenerativeNode[WO <: WorldObject]
(level: Level) extends Node(level) {

  def setOutFlow(initialObjectFlow: InitialGenerativeFlow[WO]) = addUpdatableFlows(initialObjectFlow)

  def getOutFlow(): InitialGenerativeFlow[WO] = getOutFlow(0).asInstanceOf[InitialGenerativeFlow[WO]]
}

File: ProtoSAS/src/main/java/dd/sas/pipeline/calculation/structures/nodes/generativenodes/InitialGenerativeNode.java
package dd.sas.pipeline.calculation.structures.nodes.generativenodes;

import dd.sas.pipeline.calculation.structures.flows.generativeflows.InitialObjectFlow;
import dd.sas.pipeline.calculation.structures.nodes.Node;
import dd.sas.pipeline.worldmodel.WorldObject;

import java.util.function.UnaryOperator;

/**
 * Created by Sergey on 15.09.2016.
 */
public abstract class InitialGenerativeNode<O extends WorldObject> implements Node {

    private final InitialObjectFlow<O> outFlow;

    public InitialGenerativeNode(UnaryOperator<O> dostaff) {
        this.outFlow = new InitialObjectFlow<O>(dostaff);
    }

    @Override
    public void calculate() {
        outFlow.calculate();
    }

    public InitialObjectFlow<O> getOutFlow(){
        return outFlow;
    }
}
